workflows/tree_inference.smk

```diff
@@ -0,0 +1,153 @@
+"""Snakemake rules for the tree inference pipeline."""
+
+import json
+
+from pyggdrasil.tree_inference import McmcConfig, MoveProbConfig, MoveProbConfigOptions, McmcConfigOptions
+
+
+rule make_random_or_deep_tree:
+    """Make a tree (TreeNode) of random/deep topology and save it as JSON."""
+    input:
+        script="../scripts/make_tree.py"
+
+    wildcard_constraints:
+        tree_type = "(r|d)"
+    output:
+        tree="../data/{experiment}/trees/T_{tree_type}_{n_nodes,\d+}_{tree_seed,\d+}.json"
+    shell:
+        """
+        poetry run python {input.script} \
+            --out_dir ../data/{wildcards.experiment}/trees \
+            --seed {wildcards.tree_seed} \
+            --n_nodes {wildcards.n_nodes} \
+            --tree_type {wildcards.tree_type}
+        """
+
+rule make_star_tree:
+    """"Make a tree (TreeNode) of star topology and save it as JSON."""
```

Comment:
separate rule as a star tree needs no seed. - Hence output string is shorter. Trouble with string matching :(